### PR TITLE
Change default ASCII reader numeric types to 64 bits on all platforms

### LINKED ISSUE #4684: IPAC table read error
Hell everyone,

   I'm having a problem with reading this IPAC table. It gives me the following error: 

WARNING: OverflowError converting to IntType for column cntr, using string instead. [astropy.io.ascii.core]

   When presenting these commands:

```
>> from astropy.io import ascii as ai

> > fl = 'IPAC_Example.dat'
> > ip = ascii.Ipac('right')
> > testdat = ip.read(fl)
```

I get this error no matter how I initiate "ip" -- whether leaving it default or otherwise. This isn't the only column to which this occurred and I deleted previous ones that created the problem in case that was the issue.

Any thoughts?

-Rahul Patel

### PR DESCRIPTION
**Pull request by taldcroft**
#4684 raised the issue of how the different underlying type for `np.int` on Windows (`np.int32`) vs. other platforms (`np.int64`) can be a problem.  To solve that particular issue, and more generally ensure  consistent behavior for reading ASCII tables across platforms, this changes the default types to specifically choose the 64-bit version, e.g. `np.int64` instead of `np.int`.

This is going to be an API change for Windows users.  I think it should be acceptable, but comments otherwise are welcome.

Fixes #4684.

### CODE REVIEW

File: astropy/io/ascii/ipac.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 from collections import defaultdict, OrderedDict
 from textwrap import wrap
 from warnings import warn
+import numpy as np
```
Comment (taldcroft) [inline astropy/io/ascii/ipac.py]: Ooops, this is a leftover, will remove.

**Comment by hamogu** [comment]
This should have a Changes entry since it's an API change for some users.

**Comment by hamogu** [comment]
I'm sure there is some reason that 32 bit is the default on windows - I assume there is a performance penalty to using 64 bit data types on a 32 bit system. There are few 32 bit systems left out there, so that's probably not an issue any longer; thus, I'm fine with this change (once the tests pass).

**Comment by taldcroft** [comment]
Drat, it's failing in the fast reader.  @mdmueller - do you know offhand how to change the default int output to be int64 instead of int32 on windows in the fast reader?

**Comment by taldcroft** [comment]
@mdmueller @hamogu - Just pushed up a fix that might work.

**Pull request closed without merging**